backend/corpora/common/utils/db_utils.py

```diff
@@ -29,3 +31,25 @@ def query(self, table_args: typing.List[Base], filter_args: typing.List[bool] =
             if filter_args
             else self.session.query(*table_args).all()
         )
+
+    retry_limit = 3  # The number of times to attempt generating a uuid
+
+    def generate_id(self, table: Base, *args):
+        """
+        Generates a UUID to enter into the specified table.
+
+        :param table: The table to generate the uuid  for
+        :param arg: Additional primary keys if needed
+        :return:
+        """
+        # Generate the ID
+        retry_attempts = 0
```

Comment:
Is the retry here actually necessary? The chance of collision here based on what I've read about uuid4 in the past is extraordinarily small so it seems like unnecessary code. If there does happen to be a collision, it'll get caught at the `commit` layer at which point we bubble up the error to ask the user to re-submit.

Reply:
I'm not sure if this is how SQLAlchemy works, but wouldn't it update the existing row if it existed? Thus overwriting whatever was there before.

Reply:
So we need to explicitly call update in SQLAlchemy otherwise any conflicts will be rejected.